test/ampycloud/test_scaler.py

```diff
@@ -41,65 +53,79 @@ def test_minmax_scaling():
-    deout = minmax_scaling(out, min_range=1000, mode='descale', min_val=np.nanmin(vals),
-                           max_val=np.nanmax(vals))
-
-    # Make sure we are between 0 and 1
-    assert np.nanmax(out) <= 1
-    assert np.nanmin(out) >= 0
-    # Go further and make sure we are actually much smaller than that
-    assert np.round(np.nanmax(out)-np.nanmin(out), 1) == 0.3
-    # Check that all gets properly descaled as well
-    assert np.all(np.round(deout[~np.isnan(vals)], 5) == np.round(vals[~np.isnan(vals)], 5))
-
-def test_step_scaling():
-    """ Test the step scaling function. """
+def test_step_scale():
+    """ Test the step scale function. """
 
     steps = [10, 20]
     scales = [10, 100, 1000]
 
     vals = np.array([-1, 5, 15, 25, 35])
 
-    out = step_scaling(vals, steps=steps, scales=scales, mode='scale')
+    out = step_scale(vals, steps=steps, scales=scales, mode='do')
     assert np.all(out == np.array([-0.1, 0.5, 1.05, 1.105, 1.115]))
 
-    deout = step_scaling(out, steps=steps, scales=scales, mode='descale')
+    deout = step_scale(out, steps=steps, scales=scales, mode='undo')
     assert np.all(np.round(deout, 1) == vals)
 
     # Check the continuity condition
     vals = np.arange(0, 25000, 10)
-    out = step_scaling(vals, steps=[3000, 7000, 10000, 14000], scales=[100, 500, 50, 1000, 10],
-                       mode='scale')
+    out = step_scale(vals, steps=[3000, 7000, 10000, 14000], scales=[100, 500, 50, 1000, 10],
+                       mode='do')
     # Basic check to make sure we are always increasing
     assert np.all(np.diff(out) > 0)
     # Check for gaps by looking at the slopes
     # If a slope value were to appear only once, it would be a gap.
     assert np.all([np.count_nonzero(np.diff(out)==item) for item in np.unique(np.diff(out))])
     # Check I can undo the scaling
-    deout = step_scaling(out, steps=[3000, 7000, 10000, 14000], scales=[100, 500, 50, 1000, 10],
-                         mode='descale')
+    deout = step_scale(out, steps=[3000, 7000, 10000, 14000], scales=[100, 500, 50, 1000, 10],
+                         mode='undo')
     assert np.all(np.round(deout, 1) == vals)
 
-def test_scaling():
+def test_convert_kwargs():
+    """ Test the convert_kwargs() function """
+
+    data = np.array([1, 2, 10])
+    kwargs = {'min_range': 5}
+
+    # 'min_range' is a "user" keyword
+    out = convert_kwargs(data, fct='minmax-scale', **kwargs)
+    assert 'min_range' not in out.keys()
+    assert 'min_val' in out.keys()
+    assert 'max_val' in out.keys()
+
+    kwargs = {}
+    assert 'shift' in convert_kwargs(data, fct='shift-and-scale', **kwargs).keys()
+
+
+def test_scaled():
     """ Test the umbrella scaling function, and its ability to correctly summon the different
     scaling functions. """
 
-    out = scaling(np.ones(1), fct='const', scale=10, mode='scale')
-    assert out == 0.1
+    out = scaled(np.ones(1), fct='shift-and-scale', scale=10, mode='do')
+    assert out == 0
 
```

Comment:
```suggestion
   out = scaled(np.array((10,20)), fct='shift-and-scale', scale=10, mode='do')
   assert (out == np.array(-0.5, 0.5)).all()
```
I hope it works. I did not test it :-)

Reply:
It did not work. You need to set ``shift=15`` if you want to obtain ``(-0.5, 0.5)``. By default, ``shift is taken to be ``np.nanmax(vals)``. here is the corrected test (that was implemented):

```suggestion
    out = scaled(np.array((10, 20)), fct='shift-and-scale', scale=10, shift=15, mode='do')
    assert (out == np.array([-0.5, 0.5])).all()
```